### PR TITLE
pkg/boot/initrd: return directly when passing only one initrd

### PR DESCRIPTION
**Pull request by orangecms**
I think this can still be a little better... for it would just save one unnecessary buffer/copy

### CODE REVIEW

File: pkg/boot/initrd.go [hunk 1]
```diff
@@ -21,6 +21,10 @@ func CatInitrds(initrds ...io.ReaderAt) io.ReaderAt {
 	}
 
 	return uio.NewLazyOpenerAt(strings.Join(names, ","), func() (io.ReaderAt, error) {
+		// TODO: be smarter
+		if len(initrds) == 1 {
```
Comment (hugelgupf) [inline pkg/boot/initrd.go:25]: Why do it inside the lazy opener? I'd do it further up

**Review by hugelgupf — CHANGES_REQUESTED** [review]

File: pkg/boot/initrd.go (hunk 1 above)
Comment (orangecms) [reply pkg/boot/initrd.go:25]: Yes, exactly, that needs some test cases to be updated :) I will do that

Comment (orangecms) [reply pkg/boot/initrd.go:25]: If I move this up, i.e.,
```patch
diff --git a/pkg/boot/initrd.go b/pkg/boot/initrd.go
index 0acb49a1..fac240a0 100644
--- a/pkg/boot/initrd.go
+++ b/pkg/boot/initrd.go
@@ -15,16 +15,15 @@ import (
 // CatInitrds concatenates initrds on first ReadAt call from a list of
 // io.ReaderAts, pads them to a 512 byte boundary.
 func CatInitrds(initrds ...io.ReaderAt) io.ReaderAt {
+       if len(initrds) == 1 {
+               return initrds[0]
+       }
        var names []string
        for _, initrd := range initrds {
                names = append(names, stringer(initrd))
        }

        return uio.NewLazyOpenerAt(strings.Join(names, ","), func() (io.ReaderAt, error) {
-               // TODO: be smarter
-               if len(initrds) == 1 {
-                       return initrds[0], nil
-               }
                buf := new(bytes.Buffer)
                for i, ireader := range initrds {
                        size, err := buf.ReadFrom(uio.Reader(ireader))
```

I get:

```sh
go test ./pkg/boot/
--- FAIL: TestCatInitrds (0.00s)
    initrd_test.go:106: Cat([0xc00011e340]) = name &{yay %!s(int64=0) %!s(int=-1)}, want *strings.Reader
FAIL
FAIL	github.com/u-root/u-root/pkg/boot	0.003s
FAIL
```

And I have noooo clue :(

**Comment by rminnich** [comment]
we're not gonna live or idea on a copy of what's usually a few mb, i.e. time measured in 100s of microseconds nowadays.

Happy to reconsider this if it gets fixed, but for now, let's close it.

**Pull request closed without merging**